ffig/FFIG.py

```diff
@@ -78,14 +78,47 @@ def build_model_from_source(path_to_source, module_name):
     Returns:
     - model built from a clang.cindex TranslationUnit with a name from args
     """
-    tu = clang.cindex.TranslationUnit.from_source(
-        path_to_source, '-x c++ -std=c++14 -stdlib=libc++'.split())
+    try:
+        tu = clang.cindex.TranslationUnit.from_source(
+            path_to_source, '-x c++ -std=c++14 -stdlib=libc++'.split())
+    except TranslationUnitLoadError:
```

Comment:
This is the only contentious change for me.
Other changes are clear improvements. I'll happily approve without this change.

I don't really want to litter code with try/except as Python tends to be informative when it fails.
What we do need is a check for libclang diagnostics and an exception if we find any (not necessarily in this PR).